File: src/main/java/commands/slashadapters/adapters/FAQAdapter.java
package commands.slashadapters.adapters;

import commands.runnables.informationcategory.FAQCommand;
import commands.slashadapters.Slash;
import commands.slashadapters.SlashAdapter;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;

@Slash(command = FAQCommand.class)
public class FAQAdapter extends SlashAdapter {

    public CommandData addOptions(CommandData commandData) {
        return commandData
                .addOption(OptionType.INTEGER, "page", "Which page to view", false);
    }

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        return new SlashMeta(FAQCommand.class, collectArgs(event));
    }

}

File: src/main/java/commands/slashadapters/adapters/FAQAdapter.kt
package commands.slashadapters.adapters

import commands.runnables.informationcategory.FAQCommand
import commands.slashadapters.Slash
import commands.slashadapters.SlashAdapter
import commands.slashadapters.SlashMeta
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.CommandData

@Slash(command = FAQCommand::class)
class FAQAdapter : SlashAdapter() {

    public override fun addOptions(commandData: CommandData): CommandData {
        return commandData
            .addOption(OptionType.INTEGER, "page", "Which page to view", false)
    }

    override fun process(event: SlashCommandEvent): SlashMeta {
        return SlashMeta(FAQCommand::class.java, collectArgs(event))
    }

}